### PR TITLE
Fix non-2XX response codes not being handled correctly in sync

### PR DESCRIPTION
**Pull request by vinaysshenoy**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/util/RetrofitExtensions.kt [hunk 1]
```diff
@@ -2,8 +2,14 @@ package org.simple.clinic.util
...
 import retrofit2.HttpException
+import retrofit2.Response
 
 inline fun <reified T : Any> readErrorResponseJson(error: HttpException, moshi: Moshi): T {
   val jsonAdapter = moshi.adapter(T::class.java)
   return jsonAdapter.fromJson(error.response()!!.errorBody()!!.source())!!
 }
+
+
+fun <T> Response<T>.read(): T? {
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/util/RetrofitExtensions.kt:13]: It seems like we are force casting it to non-null in sync classes, can't we just do it in this extension?

**Review by msasikanth — APPROVED** [review]
Looks good. Left one comment PTAL

File: app/src/main/java/org/simple/clinic/util/RetrofitExtensions.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/util/RetrofitExtensions.kt:13]: This is an extension method which could also be used in non-sync contexts, so I didn't want to make the assumption for those usages as well.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/util/RetrofitExtensions.kt:13]: Got it 👍🏾

**Pull request merged**